pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -330,8 +333,8 @@ static void validateTaskConfigs(TableConfig tableConfig, Schema schema) {
         // TODO task specific validate logic should directly call to PinotTaskGenerator API
         if (taskTypeConfigName.equals(REALTIME_TO_OFFLINE_TASK_TYPE)) {
           // check table is not upsert
-          Preconditions.checkState(tableConfig.getUpsertMode() == UpsertConfig.Mode.NONE,
-              "RealtimeToOfflineTask doesn't support upsert ingestion mode!");
+          Preconditions.checkState(tableConfig.getUpsertMode() == UpsertConfig.Mode.NONE || taskTypeConfig.containsKey(
```

Comment:
my thought was that we should provide it as an API param. We might extend this in future to skip all sorts of validations, which will not be limited to things in taskConfigs.
Previously from @Jackie-Jiang  as well
```
To workaround the cases where user knows it doesn't work but still want to bypass the validation, we can add some skip validation rules as an additional parameter when updating table config/schema.
```

Reply:
oh i see. instead of coding it inside tableconfig. we prefer to ask user to provide it everytime when they invoke the API. is that correct?

Reply:
Yes, essentially we can add different skip rules to allow users to skip certain checks when adding/updating the table config

Reply:
just to make sure I understand this correctly. so the solution would be to add a query_parameter to `TableConfigsRestletResource#addConfig` and `#updateConfig` 
- for example: `/tableConfigs/myTable?skipValidation=TASK,INGEST`

and then propagate these skipValidation parameter through the validation utils ONLY when tableconfig is being modified? correct?